app/code/Magento/Catalog/Test/Unit/Model/ProductTest.php

```diff
@@ -1260,40 +1278,31 @@ public function testGetMediaGalleryImagesMerging()
 
     public function testGetCustomAttributes()
     {
-        $interfaceAttributeCode = 'price';
-        $customAttributeCode = 'color';
-
-        $this->resource
-            ->method('getCustomAttributesCodes')
-            ->willReturn([$customAttributeCode]);
-        $this->model->setData($interfaceAttributeCode, 10);
+        $priceCode = 'price';
+        $colorAttributeCode = 'color';
+        $this->getCustomAttributeCodes->expects($this->exactly(3))
+            ->method('execute')
+            ->willReturn([$colorAttributeCode]);
+        $this->model->setData($priceCode, 10);
 
         //The color attribute is not set, expect empty custom attribute array
         $this->assertEquals([], $this->model->getCustomAttributes());
 
         //Set the color attribute;
-        $initialCustomAttribueValue = "red";
```

Comment:
the variables for "initial value" and "new value" wer introduced to make connection between fixture and verification explicit and clear